Mathlib/Util/CountHeartbeats.lean

```diff
@@ -213,6 +212,21 @@ register_option linter.countHeartbeats : Bool := {
 
 namespace CountHeartbeats
 
+partial def Syntax.findAndModify (modify : Syntax → Syntax) : Syntax → Syntax
+  | stx@(Syntax.node _ ``Parser.Command.declId _) => modify stx
+  | Syntax.node info kind args => Syntax.node info kind (args.map (Syntax.findAndModify modify))
+  | stx => stx
+
+/-- Traverse the syntax and add a suffix "ₕₑₐᵣₜ" to any theorem name, i.e. to any
```

Comment:
:heart: I love it!

Reply:
By the way, you might be able to use [Lean.Syntax.foldArgs](https://leanprover-community.github.io/mathlib4_docs/Lean/Parser/Basic.html#Lean.Syntax.foldArgs) here.

Reply:
maybe, although I get confused by the signature `folgArgs (s : Syntax) (f : Syntax → β → β) (b : β) : β`.

Reply:
Looking at this again, I am not sure how to use `foldArgs` in this case either: you are also modifying the syntax "in-place", whereas I was thinking of accumulating the `declId` first (e.g. in an `Array`).